services/vs-marketplace/vs-marketplace-rating.service.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+const VsMarketplaceBase = require('./vs-marketplace-base')
+const { starRating } = require('../../lib/text-formatters')
+const { floorCount } = require('../../lib/color-formatters')
+
+module.exports = class VsMarketplaceRating extends VsMarketplaceBase {
+  static get category() {
+    return 'rating'
+  }
+
+  static get route() {
+    return {
+      base: '',
+      pattern:
+        '(vscode-marketplace|vs-marketplace)/:format(r|stars)/:extensionId',
+    }
+  }
+
+  static render({ format, averageRating, ratingCount }) {
+    const message =
+      format === 'r'
+        ? `${averageRating}/5 (${ratingCount})`
+        : starRating(averageRating)
+    return {
+      label: 'rating',
```

Comment:
The label could be moved to `defaultBadgeData`.

Reply:
I was wondering if that was a possibility (meant to test but forgot to). I wasn't sure if the fact that the base class defined `defaultBadgeData` would cause problems?

Reply:
This one would take precedence.

I didn't realize we were using it in the base class. It's one of the reason I prefer functional composition to inheritance in the services; it's easy to miss stuff like that.

Adding one here would change the label in the error case, though I think that is okay.

Reply:
Agreed! Updated accordingly